GUI/Forms/Texture.cs

```diff
@@ -1,26 +1,76 @@
+using System;
 using System.Drawing;
 using System.Drawing.Imaging;
 using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
 using System.Windows.Forms;
+using GUI.Utils;
+using SkiaSharp;
+using Channels = ValveResourceFormat.CompiledShader.ChannelMapping;
 
 namespace GUI.Forms
 {
     public partial class Texture : UserControl
     {
         private string name;
+        private SKBitmap skBitmap;
+        private readonly CancellationTokenSource cts = new();
+        private readonly object channelChangingLock = new();
 
         public Texture()
         {
             InitializeComponent();
         }
 
+        public void SetImage(SKBitmap skBitmap, string name, int w, int h)
+        {
+            this.skBitmap = skBitmap;
+            viewChannelsTransparent.Enabled = skBitmap.AlphaType != SKAlphaType.Opaque;
+            SetImage(skBitmap.ToBitmap(), name, w, h);
+        }
+
         public void SetImage(Bitmap image, string name, int w, int h)
         {
             pictureBox1.Image = image;
             this.name = name;
             pictureBox1.MaximumSize = new Size(w, h);
         }
 
+        /// <summary>
+        /// Set the image to a specific channel component. E.g. Red channel.
+        /// </summary>
+        private void SetChannels(Channels channels)
+        {
+            if (skBitmap == null)
+            {
+                return;
+            }
+
+            lock (channelChangingLock)
+            {
+                Image image = null;
+                if (channels == Channels.RGBA || (skBitmap.AlphaType == SKAlphaType.Opaque && channels == Channels.RGB))
+                {
+                    image = skBitmap.ToBitmap();
+                }
+                else
+                {
+                    var pngBytes = ValveResourceFormat.IO.TextureExtract.ToPngImageChannels(skBitmap, channels);
+                    image = Image.FromStream(new MemoryStream(pngBytes));
```

Comment:
So this could be better. The Skia bitmap is kept as reference just for this `ToPngImageChannels` method, which also wastes time with the png encode.

Reply:
I tried doing a custom draw of the image with a color matrix, and it worked almost the same way, except some images would lose color data due to pre-multiplied alpha 😞